string encoder_name1
int64 left_enc

string encoder_name2
int64 right_enc
